criu/cr-dump.c

```diff
@@ -946,6 +946,7 @@ static int dump_signal_queue(pid_t tid, SignalQueueEntry **sqe, bool group)
 			} else
 				pr_perror("ptrace");
 
+			xfree(si);
```

Comment:
there are two other places where we need to call xfree(si).

Reply:
I already looked at this one a lot. The two other places are actually only one place because the first break (line 970) will also go through the second break (line 982). The reason why I did not include it is that the for loop (line 963) can end any time but the pointer to `si` is being changed by the loop. As this seems to be error code path it means we should probably run `free()` on the original pointer and not on the one after `si++`, right?